internal/wire/parse.go

```diff
@@ -813,13 +813,16 @@ func processStructProvider(fset *token.FileSet, info *types.Info, call *ast.Call
 		Out:      []types.Type{structPtr.Elem(), structPtr},
 	}
 	if allFields(call) {
-		provider.Args = make([]ProviderInput, st.NumFields())
+		provider.Args = make([]ProviderInput, 0)
 		for i := 0; i < st.NumFields(); i++ {
+			if isPrevented(st, i) {
```

Comment:
Does this logic need to be replicated in `processStructLiteralProvider`?

Reply:
Discussed offline, we decide not to add this logic to the deprecated function: first when user upgrade wire to use this new feature, they are going to make changes to their injector, it would be reasonable for them to also change the syntax on `Struct`, and secondly we stop testing the old syntax for these changes, added comment.